src/sonic-eventd/tools/events_monit_test.py

```diff
@@ -0,0 +1,236 @@
+test_event_key = "{}:{}".format(test_source, test_event_tag)
+test_event_params = {
+    "sender": os.path.basename(__file__),
+    "reason": "monit periodic test",
+    "batch-id": str(uuid.uuid1()),
+    "index": "0"
+}
+
+# Async connection wait time in milliseconds.
+ASYNC_CONN_WAIT = 300
+
+
+# Thread results
+rc_test_receive = -1
+
+
+def _log_msg(lvl, pfx, msg):
+    if lvl <= chk_log_level:
+        caller = getframeinfo(stack()[2][0])
+        fmsg = "{}:{}:{}".format(caller.function, caller.lineno, msg)
+        print("{}: {}".format(pfx, fmsg))
+        syslog.syslog(lvl, fmsg)
+
+def log_err(m):
+    _log_msg(syslog.LOG_ERR, "Err", m)
+
+def log_info(m):
+    _log_msg(syslog.LOG_INFO, "Info",  m)
+
+def log_notice(m):
+    _log_msg(syslog.LOG_NOTICE, "Notice", m)
+
+def log_debug(m):
+    _log_msg(syslog.LOG_DEBUG, "Debug", m)
+
+
+def map_dict_fvm(s, d):
+    for k, v in s.items():
+        d[k] = v
+
+
+# Invoked in a separate thread
+def test_receiver(event_obj, cnt):
+    global rc_test_receive
+
+    sh = events_init_subscriber()
+
+    # Sleep ASYNC_CONN_WAIT to ensure async connectivity is complete.
+    time.sleep(ASYNC_CONN_WAIT/1000)
+
+    exp_params = dict(test_event_params)
+
+    event_obj.set()
+    cnt_done = 0
+
+    for i in range(cnt):
+        p = event_receive_op_t()
+        rc = event_receive(sh, p)
+
+        if (rc != 0):
+            log_notice("Failed to receive. {}/{} rc={}".format(i, cnt, rc))
+            break
+
+        if test_event_key != p.key:
+            log_notice("key mismatch {} != {} {}/{}".format(test_event_key,
+                p.key, i, cnt))
+            break
+
+        exp_params["index"] = str(i)
+        rcv_params = {}
+        map_dict_fvm(p.params, rcv_params)
+
+        for k, v in exp_params.items():
+            if k in rcv_params:
+                if (rcv_params[k] != v):
+                    log_notice("key:{} exp:{} != exist:{}".format(
+                        k, v, rcv_params[k]))
+                    rc = -1
+            else:
+                log_notice("key:{} is missing", k)
+                rc = -1
+
+        if (rc != 0):
+            log_notice("params mismatch {}/{}".format(i,cnt))
+            break
+
+        if p.missed_cnt != 0:
+            log_notice("Expect missed_cnt {} == 0 {}/{}".format(p.missed_cnt,i,cnt))
+            break
+
+        if p.publish_epoch_ms == 0:
+            log_notice("Expect publish_epoch_ms != 0 {}/{}".format(i,cnt))
+            break
+
+        cnt_done += 1
+        log_debug("Received {}/{}".format(i+1, cnt))
+
+    if (cnt_done == cnt):
+        rc_test_receive = 0
+    else:
+        log_notice("test receive abort {}/{}".format(cnt_done, cnt))
+
+    # wait for a max of 5 secs for main thread to clear the event.
+    tout = 5000 
+    while(event_obj.is_set()):
+        # main thread yet to consume last set event
+        if tout > 0:
+            t_sleep = 100
+            time.sleep(t_sleep / 1000)
```

Comment:
>t_sleep [](http://example.com/codeflow?start=23&length=7)

Why 5 is long enough? Is it too long? Is there better way to determine the timing?

Reply:
Removed